python/src/tests/tasks/test_differential_expression.py

```diff
@@ -30,123 +39,153 @@ def get_request(
-        )._format_request()
-
-        baseCells = request["baseCells"]
-        backgroundCells = request["backgroundCells"]
-
-        # Check there is only one cell in each set
-        assert len(baseCells) == 1
-        assert len(backgroundCells) == 2
-
-
-    @responses.activate
-    def test_default_comparison_type_added_to_request(self, mock_S3_get):
+    def get_s3_stub(self, content_type):
+        s3 = boto3.client("s3", **config.BOTO_RESOURCE_KWARGS)
+        response = {
+            "ContentLength": 10,
+            "ContentType": "utf-8",
+            "ResponseMetadata": {
+                "Bucket": config.CELL_SETS_BUCKET,
+            },
+        }
 
-        request = DifferentialExpression(
-            self.get_request(
-                cellSet="cluster1",
-                compareWith="cluster2",
-                basis="basisCluster",
-            )
-        )._format_request()
+        expected_params = {
+            "Bucket": config.CELL_SETS_BUCKET,
+            "Key": config.EXPERIMENT_ID,
+        }
+        stubber = Stubber(s3)
+        stubber.add_response("head_object", response, expected_params)
 
-        # Check that comparisonType defaults to within
-        comparisonType = request["comparisonType"]
-        assert comparisonType == "within"
+        content_bytes = json.dumps(cell_set_types[content_type], indent=2).encode(
+            "utf-8"
+        )
 
+        data = io.BytesIO()
+        data.write(content_bytes)
+        data.seek(0)
 
-    @responses.activate
-    def test_specified_comparison_type_added_to_request(self, mock_S3_get):
+        response = {
+            "ContentLength": len(content_bytes),
+            "ContentType": "utf-8",
+            "Body": data,
+            "ResponseMetadata": {
+                "Bucket": config.CELL_SETS_BUCKET,
+            },
+        }
+        stubber.add_response("get_object", response, expected_params)
+        return (stubber, s3)
 
-        request = DifferentialExpression(
-            self.get_request(
-                cellSet="cluster1",
-                compareWith="cluster2",
-                basis="basisCluster",
-                comparisonType="between"
-            )
-        )._format_request()
+    def test_throws_on_missing_parameters(self):
+        with pytest.raises(TypeError):
+            DifferentialExpression()
 
-        # Check that comparisonType uses set value of between instead of default (within)
-        comparisonType = request["comparisonType"]
-        assert comparisonType == "between"
\ No newline at end of file
+    def test_throws_when_second_cellset_missing(self):
+        stubber, s3 = self.get_s3_stub("one_set")
+
+        with mock.patch("boto3.client") as n, stubber:
+            n.return_value = s3
+            with pytest.raises(
+                Exception, match="No cell id fullfills the 2nd cell set"
+            ):
+                DifferentialExpression(self.get_request())._format_request()
```

Comment:
is this properly linted? Looks weird in terms of indentation and new lines

Reply:
yup, the linter likes it

Reply:
this is because the Exception line is too long (longer than 80 chars) and the linter has separated it out